src/app/packages/core/utils/limited-array.ts

```diff
@@ -0,0 +1,45 @@
+import { cloneDeep } from 'lodash';
+
+export interface LimitedArray {
+	data: Array<any>;
+	limited: number;
+}
+
+// input: array, number and (*sorting function)
+// output: LimitedArray object with data property (array values up to limit) and number (how many data cells where
+// sliced off thr original array)
+// when (*sorting function) provided, sorting will be perform before slice
+export function limitArray(arr: Array<any>, limit: number, sortFn?: (a: any, b: any) => number): LimitedArray {
+	const limited = (limit < arr.length) ? arr.length - limit : 0;
+	let data = cloneDeep(arr);
+	if (sortFn) {
+		data = data.sort(sortFn);	// sort before limit
```

Comment:
The sort method sorts the Array object in place; no new Array object is created during execution.

Reply:
1) sortFn is optional.
2) It doesn't make assumption about the input (data might be in different order)